Mathlib/Topology/AlexandrovDiscrete.lean

```diff
@@ -51,3 +54,108 @@ lemma isClopen_sUnion (hS : ∀ s ∈ S, IsClopen s) : IsClopen (⋃₀ S) :=
 
 lemma isClopen_iUnion (hf : ∀ i, IsClopen (f i)) : IsClopen (⋃ i, f i) :=
 ⟨isOpen_iUnion λ i ↦ (hf i).1, isClosed_iUnion λ i ↦ (hf i).2⟩
+
+lemma interior_iInter (f : ι → Set α) : interior (⋂ i, f i) = ⋂ i, interior (f i) :=
+(interior_maximal (iInter_mono λ _ ↦ interior_subset) $ isOpen_iInter λ _ ↦
+  isOpen_interior).antisymm' $ subset_iInter λ _ ↦ interior_mono $ iInter_subset _ _
+
+lemma interior_sInter (S : Set (Set α)) : interior (⋂₀ S) = ⋂ s ∈ S, interior s := by
+  simp_rw [sInter_eq_biInter, interior_iInter]
+
+lemma closure_iUnion (f : ι → Set α) : closure (⋃ i, f i) = ⋃ i, closure (f i) :=
+compl_injective $ by simpa only [←interior_compl, compl_iUnion] using interior_iInter λ i ↦ (f i)ᶜ
+
+lemma closure_sUnion (S : Set (Set α)) : closure (⋃₀ S) = ⋃ s ∈ S, closure s := by
+  simp_rw [sUnion_eq_biUnion, closure_iUnion]
+
+end AlexandrovDiscrete
+
+variable {s t : Set α} {x y : α}
+
+/-- The *exterior* of a set is the intersection of all its neighborhoods. In an Alexandrov-discrete
+space, this is the smallest neighborhood of the set. -/
+def exterior (s : Set α) : Set α := (𝓝ˢ s).ker
+
+lemma exterior_singleton_eq_ker_nhds (a : α) : exterior {a} = (𝓝 a).ker := by simp [exterior]
+
+lemma exterior_def (s : Set α) : exterior s = ⋂₀ {t : Set α | IsOpen t ∧ s ⊆ t} := by
+  ext a
+  simp only [exterior, mem_ker, mem_sInter, mem_setOf_eq, and_imp, mem_nhdsSet_iff_forall]
+  refine' ⟨λ hs t ht hts ↦ hs _ λ b hb ↦ ht.mem_nhds $ hts hb, λ hs t ht ↦ interior_subset $
+    hs (interior t) isOpen_interior _⟩
+  simpa only [←mem_interior_iff_mem_nhds] using ht
```

Comment:
Could you reformulate [Filter.HasBasis.sInter_sets](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Order/Filter/Bases.html#Filter.HasBasis.sInter_sets) in terms of `Filter.ker` and use that here?